entity_extractor.py

```
@@ -0,0 +1,342 @@
+            )
+            for e in data.get("entities", [])
+            if e.get("name")
+        ]
+        relations = [
+            Relation(source=r["source"], target=r["target"], relation_type=r.get("type", "RELATED_TO"))
+            for r in data.get("relations", [])
+            if r.get("source") and r.get("target")
+        ]
+        return ExtractionResult(entities=entities, relations=relations)
+
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)